src/Compilers/CSharp/Test/Syntax/Parsing/StatementParsingTests.cs

```diff
@@ -3021,6 +3021,18 @@ public void ParseSubsequentElseWithoutPrecedingIfStatement()
         public void ParseElseKeywordPlacedAsIfEmbeddedStatement()
         {
             UsingStatement("if (a) else {}",
+                // (1,8): error CS1003: Syntax error, 'if' expected
```

Comment:
IMO interesting (potentially misleading). I understand the original issue was about lacking *missing `if`* indication for misplaced `else`s. This one however for `"if (a) else {}"` outputs `// (1,8): error CS1003: Syntax error, 'if' expected` as the first diagnostic. Isn't this sth unwanted?

As you this was added by following the last suggestion by @CyrusNajmabadi

Reply:
We could customize the error message here.  EatToken takes an optional error message.  If we see an 'else' we could just call `EatToken(SyntaxKind.IfToken, specialId);` where 'specialId' is the resource id for "'else' unexpected" or something like that.

Reply:
or: "'else' cannot start a statement"

Reply:
Ok, I'll add the message

Reply:
What is wrong with this error message? It sounds pretty clear to me.

Reply:
Ah I see, the example is `if (a) else { }`? That is indeed odd (even though it makes sense). I would expect something like "missing statement".